src/libcore/iter/mod.rs

```diff
@@ -298,7 +298,21 @@
 //!
 //! This will print the numbers `0` through `4`, each on their own line.
 //!
+//! Bear in mind that methods on infinite iterators, even those for which a
+//! result can be computed in finite time, may not terminate. Specifically,
```

Comment:
I think the words "can be computed" lead to a "well, how do I compute it then?" feeling.  Perhaps it could be phrased along the lines of "even if a value for them could be determined mathematically", or something?

And instead of "may not terminate", what about being more explicit, with something like "will not return successfully"?  (I don't know if talking about the different ways something could diverge would be useful.)